x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/droppable/on_drop_handler.ts

```diff
@@ -17,14 +17,12 @@ import {
 } from '../../../types';
 import {
```

Comment:
The key change in this file is getting rid of code that removes the source dimension during a drop operation. The layer panel now handles this by way of the `removeDimension` action. 

This allows `syncLinkedDimensions` to be run properly during a drag-drop sequence.